platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/SimpleListItem.kt

```diff
@@ -12,23 +12,218 @@ import androidx.compose.runtime.Composable
+ *
+ * The text will only take up one line and is ellipsized if too long to fit. The item will draw a background based on
+ * the [state]. For more granular control, see the overload with 'selected' and 'active'.
+ *
+ * @param state The state of the list item, containing selection and activity status.
+ * @param colorFilter The color filter to apply to the icon, if present.
+ * @param modifier Modifier to apply to the root container of the list item.
+ * @param iconModifier Modifier to apply to the icon displayed in the list item.
+ * @param icon An optional icon key that defines which icon should be displayed.
+ * @param iconContentDescription The content description for the icon, used for accessibility.
+ * @param style The style of the list item, including colors and metrics, with a default value from the theme.
+ * @param height The height of the list item, with a default value from global metrics in the theme.
+ * @param content The composable content to display inside the list item.
+ * @param painterHints Variadic arguments of painter hints to modify the behavior of the icon's painter.
+ */
+@Composable
+public fun SimpleListItem(
+    state: ListItemState,
+    colorFilter: ColorFilter?,
+    modifier: Modifier = Modifier,
+    iconModifier: Modifier = Modifier,
+    icon: IconKey? = null,
+    iconContentDescription: String? = null,
+    style: SimpleListItemStyle = JewelTheme.simpleListItemStyle,
+    height: Dp = JewelTheme.globalMetrics.rowHeight,
+    content: @Composable () -> Unit,
+    vararg painterHints: PainterHint,
+) {
+    Row(
+        modifier =
+            modifier
+                .semantics { selected = state.isSelected }
+                .fillMaxWidth()
+                .height(height)
+                .padding(style.metrics.outerPadding)
+                .background(
+                    color = style.colors.backgroundFor(state).value,
+                    shape = RoundedCornerShape(style.metrics.selectionBackgroundCornerSize),
+                )
+                .padding(style.metrics.innerPadding),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(style.metrics.iconTextGap),
+    ) {
+        if (icon != null) {
+            Icon(
+                modifier = iconModifier.size(16.dp),
+                key = icon,
+                contentDescription = iconContentDescription,
+                colorFilter = colorFilter,
+                hints = painterHints,
+            )
+        }
+        content()
+    }
+}
+
 /**
  * A simple list item layout comprising of a content slot and an optional icon to its start side.
  *
  * The text will only take up one line and is ellipsized if too long to fit. The item will draw a background based on
  * the [isSelected] and [isActive] values.
  */
+@Deprecated("Use the overload with selected, active, colorFilter and hints")
+@ScheduledForRemoval(inVersion = "Before 1.0")
```

Comment:
In 2025.3, I'd say :) So we have an easier time figuring things out for the next release's cleanup pass